news/2020-09-15-CDI_Lite_extension.adoc

```diff
@@ -0,0 +1,285 @@
+You can also declare all the parameters that give you access to `ClassInfo`, `MethodInfo` and `FieldInfo`, as described above, including `AppArchive`. What’s more interesting, you can also inspect existing beans and observers in the application. This is very similar to inspecting classes, methods and fields, so let’s take it quickly.
+You can declare a parameter of type `Collection<BeanInfo<? super MyService>>` to obtain information about all beans in the application that have `MyService` or any of its supertypes as one of the bean types. (Note that this example is not very useful, as `Object` is one of the supertypes of `MyService`, and all beans typically have `Object` as one of their types.) Similarly, you can declare a parameter of type `Collection<ObserverInfo<? extends MyEvent>>` to obtain information about all observers in the application that observe `MyEvent` or any of its subtypes. All the other combinations are of course also possible, and if that is not enough, there’s `AppDeployment`, which gives you more powerful querying features, similarly to `AppArchive`. You can find beans based on their scope, types, qualifiers, or the declaring class. Similarly with observers, you can filter on the observed type, qualifiers, or the declaring class.
+
+=== @Validation
+
+The most important parameter type you can declare for extensions in this phase is `Errors`. It allows you to add custom validation errors.
+What can you validate? Pretty much anything. You can get access to classes, methods and fields, just like in the `@Enhancement` phase, and you can also get access to beans and observers, just like in the `@Synthesis` phase. This includes both the `Collection<SomethingInfo<...>>` approach, and `AppArchive` / `AppDeployment` way.
+Error messages can be simple `String`s, optionally accompanied by a `DeclarationInfo`, `BeanInfo` or `ObserverInfo`, or arbitrary `Exception`s.
+In case a validation error is added, the container will prevent the application from successfully deploying (or even building, in case of build time implementations).
+
+== Conclusion
+
+You have just finished a deep dive into our current Build Compatible Extensions API proposal. Together with the API proposal, we also developed a proof-of-concept implementation in Quarkus, so that we know this API can be implemented. This proof of concept focuses solely on the `@Enhancement` phase, but that should be enough for now.
+We’re publishing the Quarkus fork in the form of a GitHub repository so that you can also experiment with it. Please bear in mind that the POC implementation is very rough and definitely is not production ready. It should be enough to evaluate the API proposal, though. Here’s how you can get your hands on it:
+
+[source,console]
+----
+git clone https://github.com/Ladicek/quarkus-fork.git
+cd quarkus-fork
+./mvnw -Dquickly
```

Comment:
You need to checkout the experiment-cdi-lite-ext branch or else this fails when you try to build it:
Scotts-iMacPro:quarkus-fork starksm$ ./mvnw -Dquickly
-bash: ./mvnw: No such file or directory

Reply:
Yeah, good point, updated it with a step to checkout the branch.

Reply:
Sorry you had to find the hard way. I originally had the `experiment-cdi-lite-ext` branch set up as the default branch of the repo, but I changed that config few days before leaving for vacation and forgot to update the document :-)